### PR TITLE
Fix XAI explanations reliability for out-of-distribution inputs (#1041)

### LINKED ISSUE #1041: Bug: XAI explanations unreliable for out-of-distribution inputs without validation
## Problem Statement

Xaytheon AI explainability system generates explanations for out-of-distribution inputs without warnings or validation checks. When users input data that looks different from the training data, the model produces confident explanations that are statistically unreliable and potentially dangerous in high-stakes applications (healthcare, finance, lending decisions).

**Business Impact:** Users trust unreliable explanations, make wrong decisions, potential legal liability if decisions cause harm.

## Root Cause Analysis

No distribution shift detection:
1. Model trained on specific data distribution
2. User inputs data outside that distribution
3. Model predicts with confidence on unfamiliar patterns
4. Explanations presented as reliable (they're not)

Example: Model trained on 25-65 age demographic, receives 90-year-old. Explains prediction as if confident, but outside training distribution.

## Solution Overview

**1. Implement Distribution Anomaly Detection**
```python
from sklearn.ensemble import IsolationForest
import numpy as np

# Train on training data distribution
anomaly_detector = IsolationForest(contamination=0.01)
anomaly_detector.fit(training_data)

# Check if new input is in distribution
def is_in_distribution(user_input, threshold=-0.5):
    score = anomaly_detector.predict(user_input)
    return score >= threshold

# Before generating explanation
if not is_in_distribution(user_input):
    return {"warning": "Input outside training distribution. Explanation unreliable."}
```

**2. Compare Distributions**
- Compare input distribution to training distribution
- Flag suspicious inputs
- Calculate distance metrics (Wasserstein, KL divergence)

**3. Confidence Calibration**
- Reduce confidence score for out-of-distribution inputs
- Attach uncertainty estimates
- Only explain when confidence > threshold AND in-distribution

**4. User Notification**
- Warn users: "This explanation may be unreliable (unusual input)"
- Show confidence level
- Suggest collecting more similar data

## Type of Change

- [x] Bug fix (explainability reliability)

## Testing Done

**Test Case 1: Out-of-distribution Detection**
- Train model on ages 25-65
- Input age 90
- Expected: Warning "outside training distribution"
- Actual (Before): Confident explanation given (unreliable)

## Related Issue

XAI reliability, model robustness, user trust in AI.

## Suggested Labels

bug, level:intermediate, gssoc26, xai-safety, distribution-shift, explainability, ai-reliability

### PR DESCRIPTION
**Pull request by gayatripadalia**
## Summary

This PR adds a validation layer for out-of-distribution (OOD) inputs in the XAI explanation workflow.

## Changes Made

* Added `isOutOfDistribution()` input validation.
* Added OOD checks before generating explanations.
* Added confidence indicators (`high` / `low`).
* Added user-facing warning banner for unreliable explanations.
* Prevented explanations from being presented as fully reliable when inputs fall outside the expected range.
* Connected warning display through `runPrediction()`.

### In-distribution input

Input: `50`

Expected Result:

* Explanation generated normally.
* Confidence: `high`.

### Out-of-distribution input

Input: `120`

Expected Result:

* Warning displayed:

  * "Input outside training distribution. Explanation may be unreliable."
* Confidence: `low`.

## Issue Addressed

Fixes #1041

This improves XAI reliability by notifying users when explanations are generated for inputs outside the expected training distribution and reducing the risk of misleading explanations.
Screenshot (1252)
Screenshot (1254)

### CODE REVIEW

**Comment by Saatvik-GT** [comment]
cool, thanks
